openeo/testing/results.py

```diff
@@ -230,9 +233,17 @@ def _compare_xarray_dataarray(
         issues.append(f"Shape mismatch: {actual.shape} != {expected.shape}")
     compatible = len(issues) == 0
     try:
-        xarray.testing.assert_allclose(a=actual, b=expected, rtol=rtol, atol=atol)
+        if (pixel_tolerance > _DEFAULT_PIXELTOL) and compatible:
+            _bad_pixels = (actual * 1.0 - expected * 1.0) > atol
```

Comment:
You should also take absolute value before comparing I think
and `rtol` should be involved too

Also: no need for the underscore in front of `_bad_pixels`  local variables are "private" per definition